nixos/modules/services/networking/networkmanager.nix

```diff
@@ -140,6 +144,34 @@ in {
         '';
       };
 
+      ethernet.macAddress = mkOption {
+        type = types.nullOr (types.either types.str (types.enum ["permanent" "preserve" "random" "stable"]));
+        default = null;
+        example = "00:11:22:33:44:55";
+        description = ''
+          "XX:XX:XX:XX:XX:XX": MAC address of the interface.
+          <literal>permanent</literal>: use the permanent MAC address of the device.
+          <literal>preserve</literal>: don’t change the MAC address of the device upon activation.
+          <literal>random</literal>: generate a randomized value upon each connect.
+          <literal>stable</literal>: generate a stable, hashed MAC address.
+          See <link xlink:href="https://blogs.gnome.org/thaller/2016/08/26/mac-address-spoofing-in-networkmanager-1-4-0/">MAC Address Spoofing in NetworkManager 1.4.0 – Thomas Haller's Blog</link> for more information.
+        '';
+      };
+
+      wifi.macAddress = mkOption {
+        type = types.nullOr (types.either types.str (types.enum ["permanent" "preserve" "random" "stable"]));
```

Comment:
Looks like the `type` is exactly the same as for the above option; perhaps it could be factored out and re-used.